### PR TITLE
Updating ELB facts module to boto3, adding support for tag filtering.

### PR DESCRIPTION
**Pull request by shousper**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

ec2_elb_facts
##### ANSIBLE VERSION

```
ansible 2.0.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

Adds support for filtering by tags on ELBs and built-in boto3 parameters for paging, etc.

Doesn't include policy information obtained via `describe_load_balancer_policies` as that didn't seem useful at this point. It probably belongs in a separate facts module OR perhaps implemented in conjunction with a boolean flag or two (likewise for policy type information).

_tl;dr: Essentially stitches together `describe_load_balancers`, `describe_instance_health`, `describe_load_balancer_attributes`, and `describe_tags` from boto3 `elb`._

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @shousper. To the current maintainers, @mjschultz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by mjschultz** [comment]
According to the [Ansible Amazon module guidelines](https://github.com/ansible/ansible-modules-extras/blob/84f2aa6167f2b8e537990067169929ca8823fb31/cloud/amazon/GUIDELINES.md#getting-started):

> Backward compatibility of older modules must be maintained.

It doesn't look like this maintains backward compatibility right now.

As far as the new feature, I think the Ansible form would be something like:

```
filters:
  tag:
    Role: nginx
    Environment: qa
```

It looks like their are some [Helper functions](https://github.com/ansible/ansible-modules-extras/blob/84f2aa6167f2b8e537990067169929ca8823fb31/cloud/amazon/GUIDELINES.md#helper-functions) for this as well. (I might be mistaken about the format.) It might be even better to have a higher level parameter that is just:

```
tags:
  - Role: nginx
  - Environment: qa
```

So, this is under needs_revision for now I think.

**Comment by gregdek** [comment]
Thanks @shousper for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@shousper A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@shousper Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@shousper A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@shousper Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@shousper A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by shousper** [comment]
Others appear to be working on better solutions. I'll close this off.
